crates/swift-integration-tests/src/option.rs

```diff
@@ -178,6 +191,20 @@ impl OptTestOpaqueRustType {
     }
 }
 
+pub struct OptTestOpaqueRustTypeRef {
+    field: u8,
+}
+
+impl OptTestOpaqueRustTypeRef {
+    fn new(field: u8) -> &'static Self {
+        &Self { field, }
```

Comment:
This won't compile since we're trying to return a reference to a temporary value.

Can do something like:

```rust
pub struct OptTestOpaqueRustTypeRef {
    field: u8,
}

impl OptTestOpaqueRustTypeRef {
    fn new_some(field: u8) -> Option<Self> {
        Some(Self { field })
    }

    fn field(&self) -> u8 {
        self.field
    }
}

fn rust_convert_option_ty_to_option_ref_ty(val: Option<OptTestOpaqueRustTypeRef>) -> Option<&OptTestOpaqueRustTypeRef> {
    val.as_ref()
}
```

Then on the Swift side can call `OptTestOpaqueRustTypeRef.new_some` and then `rust_convert_option_ty_to_option_ref_ty` to get the `Option<&OptTestOpaqueRustTypeRef>` that you need.

---

Here's how to call an associated method such as `new_some` https://github.com/chinedufn/swift-bridge/blob/master/book/src/bridge-module/functions/README.md?plain=1#L146-L189

Reply:
Ah whoops.

This is one solution:

```rust
fn rust_convert_ref_ty_to_option_ref_ty(val: &OptTestOpaqueRustTypeRef) -> Option<&OptTestOpaqueRustTypeRef> {
    Some(val)
}
``

Reply:
Maybe something like this?

```rust
pub struct OptTestOpaqueRustTypeRef {
    field: Option<OptTestOpaqueRustType>,
}

impl OptTestOpaqueRustTypeRef {
    fn new(field: u8) -> Self {
        Some(Self { field: Some(OptTestOpaqueRustType::new(field)) })
    }

    fn field(&self) -> Option<OptTestOpaqueRustType> {
        self.field
    }

    fn field_ref(&self) -> &Option<OptTestOpaqueRustType> {
        &self.field
    }
}

fn rust_convert_ref_option_ty_to_option_ref_ty(
    val: &Option<OptTestOpaqueRustType>,
) -> Option<&OptTestOpaqueRustType> {
    val.as_ref()
}
```

Reply:
Only challenge is that I'm not sure if `&Option<T>` is supported yet. SO the `val: &Option<OptTestOpaqueRustType>` might not work.

I like the general direction though. What about:

```rust
pub struct OptTestOpaqueRustTypeRef {
    field: Option<OptTestOpaqueRustType>,
}

impl OptTestOpaqueRustTypeRef {
    fn new(field: u8) -> Self {
        Some(Self { field: Some(OptTestOpaqueRustType::new(field)) })
    }

    fn field_ref(&self) -> Option<&OptTestOpaqueRustType> {
        self.field.as_ref()
    }
}
```

Then the test can call `.field_ref()` and then reflect the `Option<&OptTestOpaqueRustType>`.

Reply:
Oh right - that should work! Because then the parent object still owns the inner type. I'll try that :) thanks!

Reply:
Team work made the dream work!

Reply:
Alright, got that pushed up - hitting another compiler error (that I hit before) listed below in a general comment